### ISSUE TITLE
Lint/SymbolConversion false positive for Hash keys that end with "=" 

### ISSUE THREAD

**Issue by s-mage** [rubocop-hq/rubocop#9457]
@koic 
> Can you please open a new another issue?

Sure, here you go. Ruby allows symbols that end with `=`, but gives syntax error for such Hash keys.

```
{ "one=": 1 } # [Lint/SymbolConversion] [W]  Unnecessary symbol conversion; use `one=:` instead. ...
```

```
> x = :one=
=> :one=
> { one=: 1 }
SyntaxError: unexpected ':'
{ one=: 1 }
      ^
> { :one= => 1 }
=> {:one==>1}
> { "one=": 1 }
=> {:one==>1}
```
